### DISCUSSION TITLE
how to use blocking/similarity hash functions

### DISCUSSION THREAD

**Discussion by sumaniitm** [zinggAI/zingg#741] [Q&A] [answered]
I see in the documentation, you have mentioned domain-specific blocking and similarity function
My questions are
1. If I want to use the already provided hash functions, then how do I go about it? Is there a specific config entry needed? Can you please provide an example of usage
2. If I want to add my own blocking/similarity function, how do I add it to my downloaded Zingg package

**Comment by sonalgoyal** [comment, marked as answer]
hi @sumaniitm, the ash functions that come with Zingg are pretty exhaustive and their application is learnt from the training data as the labelling progresses. You wouldnt have to do anything extra to use them, they are already on. To add your own functions, you can fork out the Zingg repo and add custom code for blocking and similarity as detailed at https://docs.zingg.ai/zingg/improving-accuracy/definingown
